### PR TITLE
Add Odroid board selection

### PR DESCRIPTION
**Pull request by GeoffClements**
# Description

This change provides a new dialog under system that will update or append `board_name=<xu4, xu3, xu3l, hc1>` to `/boot/armbianEnv.txt`. It will also change the `BOARD_NAME` value in `/etc/armbian-release`.

The option to do this will appear only if the board family is `odroidxu4`.

The returns some functionality that was lost in the upgrade to `armbian-configng`.

Issue reference:  None.
Related documentation: None

# Implementation Details

- [x] Key changes introduced by this PR
A new entry in `config.system.json` and a new file under `tools/modules/system`.

- [x] Justification for the changes
Returns lost functionality

- [x] Confirmation that no new external dependencies or modules have been introduced
Confirmed

# Documentation Summary

- [*] **Metadata Included:**  
  _Did you include the metadata (associative arrays) in the code? Ensure that metadata for modules, jobs, and runtime has been updated appropriately._
The metadata is included, but I'm not sure I got this completely right.

- [ ] **Document Generated:**  
  _Did you generate the updated documentation using `armbian-configng --doc`? Confirm if the command was run to update `README.md` and provide any relevant details._
There are no documentation changes.

# Testing Procedure

Manually tested, conditions tested were:

- Correct line added when missing
- Correct line added when already present
- Ok and Cancel actions

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have ensured that my changes do not introduce new warnings or errors
- [x] No new external dependencies are included
- [x] Changes have been tested and verified
- [x] I have included necessary metadata in the code, including associative arrays

### CODE REVIEW

File: tools/modules/system/manage_odroid_board.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
+module_options+=(
+["manage_odroid_board,author"]=""
```
Comment (igorpecovnik) [inline tools/modules/system/manage_odroid_board.sh:2]: Put yourself here.

File: tools/json/config.system.json [hunk 2]
```diff
@@ -332,6 +332,16 @@
                     "status": "Stable",
                     "author": "@viraniac @igorpecovnik",
                     "condition": "[ -d /boot/dtb/ ] && [ -f /boot/armbianEnv.txt ]"
+                },
+                {
+                    "id": "SYODD",
+                    "description": "Select Odroid board configuration",
+                    "command": [
+                        "manage_odroid_board"
```
Comment (igorpecovnik) [inline tools/json/config.system.json:340]: name function 

module_select_xuodroid

File: tools/modules/system/manage_odroid_board.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+["manage_odroid_board,feature"]="Odroid board"
+["manage_odroid_board,desc"]="Select optimised Odroid board configuration"
+["manage_odroid_board,example"]="manage_odroid_board"
+["manage_odroid_board,status"]="review"
+)
+#
+# @description Select optimised board configuration
+#
+function manage_odroid_board() {
+	local board_list=("Odroid XU4" "Odroid XU3" "Odroid XU3 Lite" "Odroid HC1/HC2")
```
Comment (igorpecovnik) [inline tools/modules/system/manage_odroid_board.sh:13]: Keep indentation the same everywhere, similar to template
https://github.com/armbian/configng/blob/main/tools/template/module_software_template.sh

Also in module_options.

**Review by igorpecovnik — CHANGES_REQUESTED** [review]

**Review by igorpecovnik — APPROVED** [review]

**Comment by GeoffClements** [comment]
Apologies for the commits and reverts. I didn't realise those commits would appear in the PR, consequently I reverted them.

**Comment by igorpecovnik** [comment]
No worries. I think we are good to go at my last commit. Everyone had troubles with Git-Fu, if you struggle with this, I will it in the morning.

**Pull request merged**